worldengine/simulations/precipitation.py

```diff
@@ -40,20 +45,46 @@ def _calculate(seed, width, height):
         octaves = 6
         freq = 64.0 * octaves
 
+        n_scale = 1024 / float(height) #This is a variable I am adding. It exists
+                                       #so that worlds sharing a common seed but
+                                       #different sizes will have similar patterns
+
         for y in range(height):#TODO: numpy
             y_scaled = float(y) / height
-            latitude_factor = 1.0 - (abs(y_scaled - 0.5) * 2)
             for x in range(width):
-                n = snoise2(x / freq, y / freq, octaves, base=base)
+                n = snoise2((x * n_scale) / freq, (y * n_scale) / freq, octaves, base=base)
 
                 # Added to allow noise pattern to wrap around right and left.
                 if x < border:
-                    n = (snoise2(x / freq, y / freq, octaves,
+                    n = (snoise2( (x * n_scale) / freq, (y * n_scale) / freq, octaves,
                                  base=base) * x / border) + (
-                        snoise2((x + width) / freq, y / freq, octaves,
+                        snoise2(( (x * n_scale) + width) / freq, (y * n_scale) / freq, octaves,
                                 base=base) * (border - x) / border)
 
-                precipitation = (latitude_factor + n * 4) / 5.0
-                precipitations[y, x] = precipitation
+                precipitations[y, x] = n
+
+        #find ranges
+        min_precip = numpy.amin(precipitations)
+        max_precip = numpy.amax(precipitations)
+        min_temp = world.temperature['data'].min()
+        max_temp = world.temperature['data'].max()
+        precip_delta = (max_precip - min_precip)
+        temp_delta = (max_temp - min_temp)
+        
+        #normalize temperature and precipitation arrays
+        t = numpy.divide(numpy.subtract(world.temperature['data'],min_temp),temp_delta)
+        p = numpy.divide(numpy.subtract(precipitations,min_precip),precip_delta)
+        
+        #modify precipitation based on temperature
+        precipitations = numpy.add(numpy.multiply(numpy.multiply(numpy.power(t,curve_gamma),2),p),curve_bonus)
 
+        #Renormalize precipitation because the precipitation 
+        #changes will probably not fully extend from -1 to 1.
+        min_precip = numpy.amin(precipitations)
+        max_precip = numpy.amax(precipitations)
+        precip_delta = (max_precip - min_precip)
+        precipitations = numpy.subtract(numpy.divide(numpy.subtract(precipitations,min_precip),(precip_delta/2)),1)
+        min_precip = numpy.amin(precipitations)
+        max_precip = numpy.amax(precipitations)
```

Comment:
Great work on this., thanks for doing the switch yourself. :) (Also, again, just call the arrays min() and max() functions instead of using the global one).